### PR TITLE
 IBCX Adapter

### PR DESCRIPTION
**Pull request by hieuvubk**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/ibcx: 
        
 
  
    
    
    Error: Invalid balance: undefined2588179836
    Truncated error stack:
        at isValidNumber (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/generalUtil.js:119:19)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/portedTokens.js:157:14
        at Array.forEach (<anonymous>)
        at transformBalances (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/portedTokens.js:155:28)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:148:9
        at async Promise.all (index 0)
        at async Promise.all (index 2)

**Comment by hieuvubk** [comment]
This pr is  depended on https://github.com/DefiLlama/defillama-server/pull/3553 cause we r missing some tokens

**Comment by hieuvubk** [comment]
> Error while running adapter at projects/ibcx:
> 
> ```
> Error: Invalid balance: undefined2588179836
> Truncated error stack:
>     at isValidNumber (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/generalUtil.js:119:19)
>     at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/portedTokens.js:157:14
>     at Array.forEach (<anonymous>)
>     at transformBalances (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/portedTokens.js:155:28)
>     at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
>     at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:148:9
>     at async Promise.all (index 0)
>     at async Promise.all (index 2)
> ```

Cause we missing stATOM

File: projects/ibcx/index.js
[the remark this answers is not in the document]
Comment (hieuvubk) [reply projects/ibcx/index.js:1]: Thank u but why did we remove stSTAR & stJUNO cause I see stStart isnt listed on coingecko & stJUNO is really not stable. Last time I check the data from 11 days ago

**Comment by llamatester** [comment]
The adapter at projects/ibcx exports TVL: 
        
 
 
    osmosis                   105.47 k
    
    total                    105.47 k

**Comment by llamatester** [comment]
The adapter at projects/ibcx exports TVL: 
        
 
 
    osmosis                   104.52 k
    
    total                    104.52 k

File: projects/ibcx/index.js
[the remark this answers is not in the document]
Comment (g1nt0ki) [reply projects/ibcx/index.js:1]: > Thank u but why did we remove stSTAR & stJUNO cause I see stStart isnt listed on coingecko & stJUNO is really not stable. Last time I check the data from 11 days ago

my bad, when I tried running the adapter locally, I saw that they were priced. Please add them back

[the remark this answers is not in the document]
Comment (hieuvubk) [reply projects/ibcx/index.js:1]: Ah, I rechecked and it matches with data on mintscan. Thank u I think this pr ready to go.

**Comment by Define101** [comment]
Please fill the form
Or at least provide website link

**Pull request merged**